src/main/java/domain/WinningLotto.java

```diff
@@ -0,0 +1,40 @@
+package domain;
+
+import java.util.List;
+
+public class WinningLotto {
+	private final Lotto winnigLotto;
+	private final LottoBall bonusNumber;
+
+	public WinningLotto(Lotto winnigLotto, LottoBall bonusNumber) {
+		if (winnigLotto.contains(bonusNumber)) {
+			throw new IllegalArgumentException("보너스 번호는 당첨 번호와 중복되면 안됩니다.");
+		}
+		this.winnigLotto = winnigLotto;
+		this.bonusNumber = bonusNumber;
+	}
+
+	public WinningMoney getResultAmount(Lotto lotto) {
+		return getRank(lotto).getWinningMoney();
+	}
+
+	public Rank getRank(Lotto lotto) {
+		int matchNumberCount = countSameNumber(lotto);
+		if (matchNumberCount != 5) {
+			return Rank.findByMatchNumberCount(matchNumberCount);
+		}
+		if (hasBonusNumber(lotto)) {
+			return Rank.MATCH_FIVE_WITH_BONUS;
+		}
+		return Rank.MATCH_FIVE;
```

Comment:
`findByMatchNumberCount` 정적 팩토리 내부에서 전달받은 보너스볼 일치 여부 정보를 가지고 적절한 `Rank` 객체를 반환해주는 것이 더 깔끔할 것 같아요. 🙂
```suggestion
                boolean hasBonusNumber = hasBonusNumber(lotto);
		return Rank.findByMatchNumberCount(matchNumberCount, hasBonusNumber);
		
```